src/spyglass/common/common_dandi.py

```diff
@@ -0,0 +1,168 @@
+import os
+import dandi.organize
+import dandi.validate
+from dandi.validate_types import Severity
+import fsspec
+import pynwb
+import h5py
+from uuid import uuid4
+from fsspec.implementations.cached import CachingFileSystem
+
+import datajoint as dj
+
+from spyglass.common.common_usage import Export
+from spyglass.settings import export_dir
+
+from dandi.consts import known_instances
+
+dev_instance = known_instances["dandi-staging"]
+
+schema = dj.schema("common_dandi")
+
+
+@schema
+class DandiPath(dj.Manual):
+    definition = """
+    -> Export.File
+    ---
+    dandiset_id: str
+    filename: varchar(255)
+    dandi_path: varchar(255)
+    """
+
+    def fetch_file_from_dandi(self, key: dict):
+        key = (self & key).fetch1("KEY")
+        dandiset_id = str((self & key).fetch1("dandiset_id"))
+        dandi_path = (self & key).fetch1("dandi_path")
+
+        # get the s3 url from Dandi
+        from dandi.dandiapi import DandiAPIClient
```

Comment:
I keep imports at the top unless

1. There's a big import that the user would otherwise not need
2. Doing so would cause circular imports

I don't think either are the case here